src/main/java/com/google/sps/data/friend_graph/UserNode.java

```diff
@@ -59,15 +51,16 @@ public boolean equals(Object obj) {
       return false;
     }
     UserNode otherUser = (UserNode) obj;
-    return this.userID.equals(otherUser.getUserID());
+    return this.userID.equals(otherUser.userID) &&
+      this.currentUserFriendIDs.equals(otherUser.currentUserFriendIDs);
   }
 
   @Override
   public int hashCode() {
     int hashTotal = 0;
-    for (int i = 0; i < this.userID.length(); i++) {
-      hashTotal += (i * (int)this.userID.charAt(i));
-    }
+    final int prime = 31;
```

Comment:
nit: explain where the number 31 comes from in a comment

Reply:
Added a comment on this line that explains it! Main idea here was using a prime number to help calculate a more unique hash value.